scanner/src/main/kotlin/provenance/PackageProvenanceResolver.kt

```diff
@@ -52,7 +50,7 @@ interface PackageProvenanceResolver {
      *
      * Throws an [IOException] if the provenance cannot be resolved.
      */
-    fun resolveProvenance(pkg: Package, defaultSourceCodeOrigins: List<SourceCodeOrigin>): KnownProvenance
+    suspend fun resolveProvenance(pkg: Package, defaultSourceCodeOrigins: List<SourceCodeOrigin>): KnownProvenance
```

Comment:
Can callers like

https://github.com/oss-review-toolkit/ort/blob/74557ba1428f578d1456535435591292714c469d/scanner/src/main/kotlin/Scanner.kt#L245-L249

now be simplified?

Reply:
No, the `async` is used here to run multiple resolutions in parallel, the `runCatching` to catch exceptions, and the `pkg to` to make the `pkg` available for later code.